OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/FirebaseAnalytics.java

```diff
@@ -0,0 +1,378 @@
+                Values.SOCIAL_CERTIFICATE_SHARED);
+        event.putCourseId(courseId);
+        event.putString(Keys.CATEGORY, Values.SOCIAL_SHARING);
+        event.putString(Keys.URL, certificateUrl);
+        event.putString(Keys.TYPE, getShareTypeValue(method));
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackCourseComponentViewed(String blockId, String courseId) {
+        final FirebaseEvent event = new FirebaseEvent(Events.COMPONENT_VIEWED,
+                Values.COMPONENT_VIEWED);
+        event.putCourseId(courseId);
+        event.putString(Keys.BLOCK_ID, blockId);
+
+        //Add category for Google Analytics
+        event.addCategoryToBiEvents(Values.NAVIGATION, Keys.COMPONENT_VIEWED);
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackOpenInBrowser(String blockId, String courseId, boolean isSupported) {
+        final FirebaseEvent event = new FirebaseEvent(Events.OPEN_IN_BROWSER,
+                Values.OPEN_IN_BROWSER);
+        event.putCourseId(courseId);
+        event.putString(Keys.BLOCK_ID, blockId);
+        event.putBoolean(Keys.SUPPORTED, isSupported);
+
+        //Add category for Google Analytics
+        String label = (isSupported ? Values.OPEN_IN_WEB_SUPPORTED : Values.OPEN_IN_WEB_NOT_SUPPORTED);
+        event.addCategoryToBiEvents(Values.NAVIGATION, label);
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackProfileViewed(@NonNull String username) {
+        final FirebaseEvent event = new FirebaseEvent(Events.PROFILE_VIEWED, Values.PROFILE_VIEWED);
+        event.addCategoryToBiEvents(Values.PROFILE, username);
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackProfilePhotoSet(boolean fromCamera) {
+        final FirebaseEvent event = new FirebaseEvent(Events.PROFILE_PHOTO_SET,
+                Values.PROFILE_PHOTO_SET);
+        event.addCategoryToBiEvents(Values.PROFILE, fromCamera ? Values.CAMERA : Values.LIBRARY);
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    /**
+     * This function is set to identify the user for subsequent calls
+     *
+     * @param userID   - User Id from the server
+     * @param email    -  email of the user
+     * @param username -  Username/email that the user uses for signing in
+     */
+    @Override
+    public void identifyUser(String userID, String email, String username) {
+        /*Traits traits = new Traits();
+        traits.putEmail(email);
+        traits.putUsername(username);
+        tracker.identify(userID, traits, new Options());*/
+    }
+
+    /**
+     * This resets the Identify user once the user has logged out
+     */
+    @Override
+    public void resetIdentifyUser() {
+//        tracker.resetIdentifyUser();
```

Comment:
I forgot to implement this function, but I guess I'll need to call the `setUserId` function with a `null` here.